### ISSUE TITLE
Login Page FE

### ISSUE THREAD

**Issue by claytonleonardcook** [SVSU-Capstone25/PLOT-FE#13]
## 🎯 Acceptance Criteria

> [!NOTE]
> The following are developer friendly images and comments to explain what will be accepted as done.

![Image](https://github.com/user-attachments/assets/b418faf1-539b-4ff3-9080-3173d53e48ee)

![Image](https://github.com/user-attachments/assets/bdbbf8b7-30a0-4d03-a104-23fe70114704)

The user should be able to press tab to navigate through the login page. The tab order should be username input, password input and "Log In" button.

### Email Text Input

The email input field should  allow for any characters with a max of 32.

If the user inputs more than 32 characters then the input should error and show "Only a max of 32 characters are allowed in the email field."

- [ ] Create a text input.
- [ ] Error message displayed for invalid inputs

### Password Text Input

The text input should be hidden by default.

There should be a eye icon to allow users to see what they've input.

The password should be no more than 32 characters, allow for numbers, special character (!, @, $, etc.), uppercase and lowercase letters

If the user inputs more than 32 characters there should be an error message that indicates that the max character length is 32. The error should be along the lines of "Only a max of 32 characters are allowed in the password field."

If the user utilizes invalid character there should be an error message indicating it is not an accepted character. Such as "Only alpha numerical characters are allowed in the password field."

- [ ] Create a text input
- [ ] Text input only takes alpha numeric and special characters
- [ ] Password hidden by default
- [ ] Functional eye icon to view their input (This can be an after thought not required)
- [ ] Error message displayed for invalid inputs

### Forgot Password Button
There should be a button labeled "Forgot Password" that redirects the user to a password recovery page.
- [ ] Button exists
- [ ] Redirects user to recovery
### Log In Button

After the user enters their correct username, correct password and clicks the "Login" button then they should be redirected to the Store Dashboard.

The "Log In" button should be disabled if there are any errors.

After a user enters either/or an incorrect username/password, the user should remain at the login page with a displayed error of "Incorrect password"

- [ ] User redirected to store page upon successful login
- [ ] Button disabled if error detected in either username or password
- [ ] Error message displayed after wrong username/password
## 🧪 Testing

> [!NOTE]
> The following are what the E2E tests expect once development is done.

When a user hovers over the "..." on a user I **EXPECT**:
- A hover menu to pop up and show "Edit" and "Delete" buttons.

---

When a user clicks the "Delete" button I **EXPECT**:
- A modal to pop up that says "Are you sure you want to delete this user?" with a "Confirm" and "Cancel" button. 

---

... (DELETE THIS PLACEHOLDER CONTENT)

## 🔗 Links & Notes

> [!NOTE]
> The following are links and any other notes to aid other developers and leads.

- [Valid Login Page Design](https://www.figma.com/design/fqcAForJKde0zmJEAhm617/PLOT---Plato's-Layout-Optimization-Tool?node-id=193-8241&t=MVOCajM5k5cDGdzf-1)
- [Invalid Login Page Design](https://www.figma.com/design/fqcAForJKde0zmJEAhm617/PLOT---Plato's-Layout-Optimization-Tool?node-id=193-8242&t=MVOCajM5k5cDGdzf-1)
- [WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/)
